torch_geometric/datasets/web_qsp_dataset.py

```diff
@@ -140,121 +164,223 @@ class KGQABaseDataset(InMemoryDataset):
             If :obj:`"test"`, loads the test dataset. (default: :obj:`"train"`)
         force_reload (bool, optional): Whether to re-process the dataset.
             (default: :obj:`False`)
+        verbose (bool, optional): Whether to print output. Defaults to False.
         use_pcst (bool, optional): Whether to preprocess the dataset's graph
             with PCST or return the full graphs. (default: :obj:`True`)
+        use_cwq (bool, optional): Whether to load the ComplexWebQuestions dataset. (default: :obj:`True`)
+        load_dataset_kwargs (dict, optional): Keyword arguments for the `datasets.load_dataset` function. (default: :obj:`{}`)
     """
     def __init__(
-        self,
-        dataset_name: str,
-        root: str,
-        split: str = "train",
-        force_reload: bool = False,
-        use_pcst: bool = True,
-        use_cwq: bool = True,
+            self,
+            dataset_name: str,
+            root: str,
+            split: str = "train",
+            force_reload: bool = False,
+            verbose: bool = False,
+            use_pcst: bool = True,
+            use_cwq: bool = True,
+            load_dataset_kwargs: Dict[str, Any] = dict(),
     ) -> None:
+        self.split = split
         self.dataset_name = dataset_name
         self.use_pcst = use_pcst
+        self.load_dataset_kwargs = load_dataset_kwargs
+        # Caching custom subsets of the dataset results in unsupported behavior
+        if 'split' in load_dataset_kwargs:
+            print(
+                "WARNING: Caching custom subsets of the dataset results in unsupported behavior. Please specify a separate root directory for each split, or set force_reload=True on subsequent instantiations of the dataset."
+            )
+        # TODO Confirm with @riship why the dependency checks and device setting were removed here # noqa
+        '''
+        self.device = torch.device(
+            "cuda" if torch.cuda.is_available() else "cpu")
+        self._check_dependencies()
+        '''
+
+        # TODO: @riship should this be done in the base class?
+        self.required_splits = ['train', 'validation', 'test']
+
+        self.verbose = verbose
+        self.force_reload = force_reload
         super().__init__(root, force_reload=force_reload)
 
-        if split not in {'train', 'val', 'test'}:
+        # NOTE: Current behavior is to process the entire dataset, and only return the split specified by the user
```

Comment:
if you call this dataset 3 times
```
WebQSP(split="train")
..
Webqsp(..
```
it just processes it once and then returns train on the first call, but the val and test ones just pull from the already processed data